### ISSUE TITLE
Incorrect checksum for freed object 0x108040e50: probably modified after being freed, thread name = 'io.flutter.1.raster', queue = 'compiler request queue', stop reason = signal SIGABRT

### ISSUE THREAD

**Issue by git-boya** [flutter/flutter#77870]
If it's not installed, there won't be any problems on the first run,and there's no problem on the simulator

The crash happened after ```runApp(MyApp());```  and before ```initState```

flutter doctor -v
```
[✓] Flutter (Channel stable, 2.0.1, on macOS 11.2.2 20D80 darwin-x64, locale zh-Hans-CN)
    • Flutter version 2.0.1 at /Users/ieducc/develop/flutter
    • Framework revision c5a4b4029c (6 days ago), 2021-03-04 09:47:48 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/ieducc/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] Connected device (3 available)
    • MI 8 (mobile)       • 19ee4aeb                                 • android-arm64  • Android 10 (API 29)
    • 平亚彬的iPhone (mobile) • 2a23117dcd977b7461028b064d848bd15b8ef338 • ios            • iOS 14.4.1
    • Chrome (web)        • chrome                                   • web-javascript • Google Chrome 89.0.4389.82


```
flutter run --verbose
```
[        ] (lldb)     target modules search-paths add /usr "/Users/ieducc/Library/Developer/Xcode/iOS DeviceSupport/14.4.1 (18D61)/Symbols/usr" /System "/Users/ieducc/Library/Developer/Xcode/iOS DeviceSupport/14.4.1 (18D61)/Symbols/System"
"/private/var/containers/Bundle/Application/D750DBD4-1B6E-4055-A6BA-2262E7252CCF" "/Users/ieducc/develop/flutter_cloud/build/ios/iphoneos" "/var/containers/Bundle/Application/D750DBD4-1B6E-4055-A6BA-2262E7252CCF"
"/Users/ieducc/develop/flutter_cloud/build/ios/iphoneos" /Developer "/Users/ieducc/Library/Developer/Xcode/iOS DeviceSupport/14.4.1 (18D61)/Symbols/Developer"
[  +37 ms] (lldb)     command script import "/tmp/5E33DFD1-6518-4772-B17A-B9C2D82A7163/fruitstrap_2a23117dcd977b7461028b064d848bd15b8ef338.py"
[   +4 ms] (lldb)     command script add -f fruitstrap_2a23117dcd977b7461028b064d848bd15b8ef338.connect_command connect
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_2a23117dcd977b7461028b064d848bd15b8ef338.run_command run
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_2a23117dcd977b7461028b064d848bd15b8ef338.autoexit_command autoexit
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_2a23117dcd977b7461028b064d848bd15b8ef338.safequit_command safequit
[        ] (lldb)     connect
[  +41 ms] (lldb)     run
[ +206 ms] success
[        ] Application launched on the device. Waiting for observatory port.
[+1636 ms] (lldb) 2021-03-11 09:28:50.678218+0800 my app[596:43698] adding notification observers
[  +49 ms] Engine creation error: Error Domain=com.apple.CoreHaptics Code=-4809 "(null)"
[ +111 ms] Observatory URL on device: http://127.0.0.1:53970/
[   +5 ms] Attempting to forward device port 53970 to host port 53940
[        ] executing: /Users/ieducc/develop/flutter/bin/cache/artifacts/usbmuxd/iproxy 53940:53970 --udid 2a23117dcd977b7461028b064d848bd15b8ef338
[+1040 ms] Forwarded port ForwardedPort HOST:53940 to DEVICE:53970
[        ] Forwarded host port 53940 to device port 53970 for Observatory
[   +1 ms] Installing and launching... (completed in 25.9s)
[        ] Caching compiled dill
[ +184 ms] Connecting to service protocol: http://127.0.0.1:53940/
[   +9 ms] flutter: ThemeData#1ee6c
[        ] my app(596,0x16b66f000) malloc: Incorrect checksum for freed object 0x108040e50: probably modified after being freed.
[        ] Corrupt value: 0x9000000020202270
[        ] my app(596,0x16b66f000) malloc: *** set a breakpoint in malloc_error_break to debug
[  +76 ms] Process 596 stopped
[        ] * thread #10, name = 'io.flutter.1.raster', queue = 'compiler request queue', stop reason = signal SIGABRT
[        ]     frame #0: 0x00000001bd08884c libsystem_kernel.dylib`__pthread_kill + 8
[        ] libsystem_kernel.dylib`__pthread_kill:
[        ] ->  0x1bd08884c <+8>:  b.lo   0x1bd088868               ; <+36>
[        ]     0x1bd088850 <+12>: stp    x29, x30, [sp, #-0x10]!
[        ]     0x1bd088854 <+16>: mov    x29, sp
[        ]     0x1bd088858 <+20>: bl     0x1bd065f5c               ; cerror_nocancel
[        ] Target 0: (my app) stopped.
[        ] ios-deploy exited with code 0
[  +88 ms] Fail to connect to service protocol: http://127.0.0.1:53940/: HttpException: Connection closed before full header was received, uri = http://127.0.0.1:53940/ws
[ +591 ms] Error connecting to the service protocol: failed to connect to http://127.0.0.1:53940/
[   +3 ms] "flutter run" took 65,866ms.
[   +4 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:644:9)
           <asynchronous suspension>
           #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1157:12)
           <asynchronous suspension>
           #3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1009:27)
           <asynchronous suspension>
           #4      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #5      AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #6      CommandRunner.runCommand (package:args/command_runner.dart:197:13)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:278:9)
           <asynchronous suspension>
           #8      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #9      AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #10     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:234:5)
           <asynchronous suspension>
           #11     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:64:9)
           <asynchronous suspension>
           #12     run.<anonymous closure> (package:flutter_tools/runner.dart:62:12)
           <asynchronous suspension>
           #13     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #14     AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #15     runInContext (package:flutter_tools/src/context_runner.dart:73:10)
           <asynchronous suspension>
           #16     main (package:flutter_tools/executable.dart:90:3)
           <asynchronous suspension>
           
           
[ +268 ms] ensureAnalyticsSent: 253ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hook priority 4
[        ] Shutdown hooks complete
[        ] exiting with code 2

```

**Comment by pedromassangocode** [comment]
Duplicate of https://github.com/flutter/flutter/issues/77347. Please follow up and subscribe on that issue for updates.

**Comment by git-boya** [comment]
> Duplicate of #77347. Please follow up and subscribe on that issue for updates.

I tried to copy the code to a new project, and this problem seems to have been resolved

**Comment by jmagman** [comment]
@pedromassango I think that's a totally different crash.
```
[   +9 ms] flutter: ThemeData#1ee6c
[        ] my app(596,0x16b66f000) malloc: Incorrect checksum for freed object 0x108040e50: probably modified after being freed.
[        ] Corrupt value: 0x9000000020202270
[        ] my app(596,0x16b66f000) malloc: *** set a breakpoint in malloc_error_break to debug
[  +76 ms] Process 596 stopped
[        ] * thread #10, name = 'io.flutter.1.raster', queue = 'compiler request queue', stop reason = signal SIGABRT
[        ]     frame #0: 0x00000001bd08884c libsystem_kernel.dylib`__pthread_kill + 8
[        ] libsystem_kernel.dylib`__pthread_kill:
[        ] ->  0x1bd08884c <+8>:  b.lo   0x1bd088868               ; <+36>
[        ]     0x1bd088850 <+12>: stp    x29, x30, [sp, #-0x10]!
[        ]     0x1bd088854 <+16>: mov    x29, sp
[        ]     0x1bd088858 <+20>: bl     0x1bd065f5c               ; cerror_nocancel
```

**Comment by cuishijie1991** [comment]
same issue!  can be launched with Xcode not idea

**Comment by cuishijie1991** [comment]
when installed by idea, simulator and real device on 12.4.9 is ok, but  on 14.3, 14.4 is not 。 there is no other error info!
![截屏2021-03-15 下午2 12 16](https://user-images.githubusercontent.com/17976216/111110967-72a6f680-8598-11eb-9b1a-c5c7d22b4776.png)

**Comment by chinmaygarde** [comment]
@jmagman There isn't enough information here to figure out if these crashes are different. For https://github.com/flutter/flutter/issues/78076 we are attempting instrumented builds. We'll try the same builds to see if this error can be reproduced.

**Comment by jmagman** [comment]
> @jmagman There isn't enough information here to figure out if these crashes are different. 

Yup, I agree.  I didn't see any crash in the originator's report at https://github.com/flutter/flutter/issues/77347 so I didn't think it should just be duplicated without investigation.

**Comment by cuishijie1991** [comment]
I fixed my problem by upgrading my Xcode from v12.1 to v12.4 。
hope this can help people who have the same problem

**Comment by liaoshuihantan** [comment]
If your app's Product Name isn't English,try to change it with English. I fixed my problem by use English Product Name.
